syntax = "proto3";

service Customer {
  rpc CreateCustomer(CreateCustomerRequest) returns (CreateCustomerResponse) {}
  rpc ReadCustomer(ReadCustomerRequest) returns (ReadCustomerResponse) {}
  rpc ReadCustomerList(ReadCustomerListRequest)
      returns (ReadCustomerListResponse) {}
  rpc UpdateCustomer(UpdateCustomerRequest) returns (UpdateCustomerResponse) {}
  rpc DeleteCustomer(DeleteCustomerRequest) returns (DeleteCustomerResponse) {}
}

// #### Requests ####
message CreateCustomerRequest {
  int32 subscription_id = 1;
  string first_name = 2;
  string last_name = 3;
  string email = 4;
  string phone_number = 5;
  string created_at = 6;
  string updated_at = 7;
}

message ReadCustomerRequest { int32 id = 1; }

message ReadCustomerListRequest {}

message UpdateCustomerRequest {

  int32 id = 1;
  int32 subscription_id = 2;
  string first_name = 3;
  string last_name = 4;
  string email = 5;
  string phone_number = 6;
  
}

message DeleteCustomerRequest { int32 id = 1; }



// #### Response ####
message CreateCustomerResponse { string msg = 1; }

message ReadCustomerResponse {
  int32 id = 1;
  int32 subscription_id = 2;
  string first_name = 3;
  string last_name = 4;
  string email = 5;
  string phone_number = 6;
  string created_at = 7;
  string updated_at = 8;
}

message ReadCustomerListResponse {
  message CustomerObject {
    int32 id = 1;
    int32 subscription_id = 2;
    string first_name = 3;
    string last_name = 4;
    string email = 5;
    string phone_number = 6;
    string created_at = 7;
    string updated_at = 8;
  }
  repeated CustomerObject customer_list = 1;
}

message UpdateCustomerResponse {
  string msg = 1;
}

message DeleteCustomerResponse {
  string msg = 1;
}
